wgpu-core/src/binding_model.rs

```diff
@@ -406,6 +408,8 @@ pub struct PipelineLayout<B: hal::Backend> {
     pub(crate) life_guard: LifeGuard,
     pub(crate) bind_group_layout_ids: ArrayVec<[Valid<BindGroupLayoutId>; MAX_BIND_GROUPS]>,
     pub(crate) push_constant_ranges: ArrayVec<[wgt::PushConstantRange; SHADER_STAGE_COUNT]>,
+    #[cfg(debug_assertions)]
+    pub(crate) label: String,
```

Comment:
I'm wondering... if this is on all objects, and `life_guard` is in all objects, can we put the label inside it?

Reply:
At least I can't see it on `BindGroupLayout` or `CommandBuffer`. But I guess it could be moved for the ones that do have it?

Reply:
Yes. It looks like all the things that have `life_guard` also need the label, so we should move it in.
For others few we can have it aside, like your PR does now.